### ISSUE TITLE
(Bug Report) Internal Error in sir( ) ;  beta (or gamma) between 0 and 1 => Error

### ISSUE THREAD

**Issue by hitaysh** [igraph/igraph#760]
I am simulating a non-dimensional SIR model using igraph.

I run into Internal Errors when beta or gamma is a fraction between 0 and 1. Is this not allowed in iGraph? 

For example:

```
g <- erdos.renyi.game(1000, 1/1000)
sir(g,0.4,1)

# Error in .Call("R_igraph_sir", graph, beta, gamma, no.sim, PACKAGE = "igraph") : 
#  At sir.c:185 : Internal SIR error, Internal error, likely a bug in igraph
```

For my model I'll need to use following parameters:

```
b  <- 0.0006393279 # beta
gm <- 1/7          # gamma
```

Is there a way to use above parameters with `sir()` ?

**Comment by sfrosenb** [comment]
Any updates on this? I'd like to use the SIR function for teaching purposes comparing an ODE SIR model to the network version, but I cannot do that effectively if beta and gamma have to be greater than 1.

**Comment by ntamas** [comment]
Sorry, there's no progress on this issue so far.

**Comment by vtraag** [comment]
This is essentially the same problem as #620, which is (or will be) solved in #1205.
